lib/Semantics/assignment.cpp

```diff
@@ -110,16 +125,18 @@ static const char *WhyBaseObjectIsSuspicious(
   }
 }
 
-// Checks C1594(1,2)
-void CheckDefinabilityInPureScope(parser::ContextualMessages &messages,
+// Checks C1594(1,2); false if check fails
+bool CheckDefinabilityInPureScope(parser::ContextualMessages &messages,
```

Comment:
Similarly, how about calling this "IsDefinableInPureScope()"?

Reply:
It's not a pure predicate function; it has side effects.  `Check...` is fine.

Reply:
The main purpose of the function is to perform the check and emit error messages if it fails. The result is just to indicate that they passed so we can continue with other checks. So I think the current name is better.